internal/whrequest/request.go

```diff
@@ -0,0 +1,59 @@
+package whrequest
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/hatchet-dev/hatchet/internal/signature"
+)
+
+func Send(ctx context.Context, url string, secret string, data any, headers ...func(req *http.Request)) ([]byte, error) {
+	body, err := json.Marshal(data)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", url, bytes.NewReader(body))
+	if err != nil {
+		return nil, err
+	}
+
+	sig, err := signature.Sign(string(body), secret)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("X-Hatchet-Signature", sig)
+
+	for _, h := range headers {
+		h(req)
+	}
+
+	httpClient := &http.Client{
+		// use 10 minutes timeout
+		Timeout: time.Second * 600,
```

Comment:
why 10m?

Reply:
Good point – I chose this randomly because the default http client doesn't have a timeout. I figured workflows which run over http should not really take that long, but it's set completely arbitrary and if it says we should document it